src/swarm-lang/Swarm/Language/Load.hs

```diff
@@ -67,34 +116,27 @@ resolve prov s = do
         Left _ -> currentDir
         Right (loc, _) -> importDir loc
 
-  (resMap, (impSet, s')) <- runState mempty . resolveImports cur $ s
+  (resMap, (impSet, s')) <- evalState @(Set ResLoc) S.empty . runState @(SourceMap Resolved) OM.empty . resolveImports cur $ s
```

Comment:
The phantom state parameter makes this line a bit too long, how about adding a (not exported) constant:
```Haskell
emptyImports :: Set ResLoc
emptySourceMap :: SourceMap Resolved
```